audio-file-parser/src/audio_file.cpp

```diff
@@ -0,0 +1,151 @@
+#include "audio_file.h"
+
+AudioFile::AudioFile(string filename) {
+    this->filename_ = filename;
+    this->Load();
+}
+
+AudioFile::~AudioFile() {
+    this->file_.close();
+}
+
+/**
+ * Reads raw audio data from file. To support 24-bit audio, data is first
+ * read one byte at a time into uint8_t's. Using bitwise shift, the 8-bit
+ * chunks pieced together into a int32_t container. Audio data is stored 
+ * in Two's Complement, so a mask is applied conditionally applied to 
+ * preserve sign in the new container. Lastly, the integers are converted
+ * to doubles for further processing.   
+ **/
+void AudioFile::ParseAudioData() {
+    // Create temporary vector to store 8-bit chunks of audio data
+    vector<uint8_t> audio_data(this->audio_data_length_);
+    for (int i = 0; i < this->audio_data_length_; i++) {
+        uint8_t sample;
+
+        this->file_.read(reinterpret_cast<char *>(&sample), sizeof(uint8_t));
+
+        audio_data[i] = sample;
+    }
+    
+    // Initialize vector for each channel of audio data
+    for (int i = 0; i < this->num_channels_; i++) {
+        this->audio_data_.push_back(vector<double>(this->frames_));
+    }
```

Comment:
Can this be done without a loop? 
```suggestion
this->audio_data_ = (vector<vector<double>(this->frames_)>(this->num_channels_);
```

Reply:
Nope.